python/ray/data/datasource/bigquery_datasink.py

```diff
@@ -71,25 +77,35 @@ def _write_single_block(block: Block, project_id: str, dataset: str) -> None:
                 pq.write_table(block, fp, compression="SNAPPY")
 
                 retry_cnt = 0
-                while retry_cnt < MAX_RETRY_CNT:
+                while retry_cnt <= self.max_retry_cnt:
```

Comment:
When `max_retry_cnt is None` (default), where do we convert `max_retry_cnt` to an `int`?

Reply:
In BigQueryDatasink, `max_retry_cnt: Optional[int] = DEFAULT_MAX_RETRY_CNT`

Reply:
`write_bigquery` passes `None` to `BigQueryDatasink`, so the value of `max_retry_cnt` is `None`? The default `DEFAULT_MAX_RETRY_CNT` value is overwritten by `max_retry_cnt` specified in `write_bigquery`.

Reply:
Oh good catch, I see what you mean now.